Assets/Scripts/Tools/Extrude.cs

```diff
@@ -474,6 +502,32 @@ void CreateVisuals(MeshRebuilder meshRebuilder, List<Vector3> newVertices, List<
         }
     }
 
+    // Lock new visuals if any of the old vertices are locked
+    void LockNewVisuals(MeshRebuilder meshRebuilder, int lockedVertex)
+    {
+        foreach(Edge e in meshRebuilder.edgeObjects)
```

Comment:
instead, try going through the connectedFaces/connectedEdges lists on the Vertex object see if that works